x/upgrade/keeper/keeper.go

```diff
@@ -48,6 +49,39 @@ func (k Keeper) SetUpgradeHandler(name string, upgradeHandler types.UpgradeHandl
 	k.upgradeHandlers[name] = upgradeHandler
 }
 
+// SetVersionMap saves a given version map to state
+func (k Keeper) SetVersionMap(ctx sdk.Context, vm module.VersionMap) {
+	if len(vm) > 0 {
+		store := ctx.KVStore(k.storeKey)
+		versionStore := prefix.NewStore(store, []byte{types.VersionMapByte})
+		for modName, ver := range vm {
+			nameBytes := []byte(modName)
+			verBytes := make([]byte, 8)
+			binary.BigEndian.PutUint64(verBytes, ver)
+			versionStore.Set(nameBytes, verBytes)
+		}
+	}
```

Comment:
Why do these not happen in InitGenesis? I thought we agreed upon that.

Reply:
[I do that here](https://github.com/cosmos/cosmos-sdk/pull/8743/files#diff-8d1ca8086ee74e8f0490825ba21e7435be4753922192ff691311483aa3e71a0aR424)

Reply:
this seemed like correct spot based on the suggestions, did you have somewhere else in mind?

Reply:
Maybe you're right. I guess I was thinking it could be in InitGenesis but maybe it doesn't matter